src/lib/profiles/service-directory/ServiceDirectory.cpp

```diff
@@ -538,13 +551,61 @@ WEAVE_ERROR WeaveServiceManager::connect(uint64_t aServiceEp,
     return err;
 }
 
+/**
+ * @brief This method looks up directory information for a service endpoint.
+ *
+ * If the service directory has been resolved, i.e. if there has been a
+ * successful connect() operation, then this method will populate the supplied
+ * HostPortList object.
+ *
```

Comment:
Maybe add a note that the returned HostPortList contains pointers into the service directory cache, and thus only remains valid until the cache is cleared or another service directory lookup occurs.

Reply:
Done.